test/TestProjects/Models-Cadl/Generated/BaseModelWithDiscriminator.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace ModelsInCadl
+{
+    /// <summary> Base model with discriminator property. </summary>
+    public partial class BaseModelWithDiscriminator
```

Comment:
The 'discriminatorProperty' is missing in the base model. The discriminator is an field of the base model.

Reply:
discriminator support is not implemented in autorest.csharp in this PR. That would be a separate PR after we close several open issues:
- should base class with discriminator be abstract?
- do we need discriminator property or constant is enough?
- what should we do if discriminator property value is different from anything we handle?

Reply:
FYI mgmt plane decides the base class to be abstract: https://github.com/Azure/autorest.csharp/issues/2430
Should we align?

Reply:
discriminator is not in the scope of for August MVP so can we leave as it is for now?  @annelo-msft 
This part is likely to be redesigned later.